packages/openspp/src/Adaptor.js

```diff
@@ -918,6 +918,146 @@ export function removeFromGroup(group_id, individual_id) {
+ * @example <caption>search with offset</caption>
+ * searchServicePoint([["name", "ilike", "agent 1"]], 3)
+ * @function
+ * @param {Array} domain - searching domain
+ * @param {number} [offset=0] - offset searching
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function searchServicePoint(domain, offset=0, callback=(s) => s) {
+  return async state => {
+    try {
+      domain = resolveDomain(domain);
+        let servicePoints = await sppConnector.searchRead(
+        'spp.service.point',
+        domain,
+        [
+          'name',
+          'area_id',
+          'service_type_ids',
+          'program_id',
+          'phone_sanitized',
+          'is_contract_active',
+          'is_disabled',
+        ],
+        {
+          limit: 100,
+          offset: offset,
+          order: 'id desc'
+        },
+      );
+      for (let servicePoint of servicePoints) {
+        servicePoint.program_ids = servicePoint.program_id;
+        delete servicePoint.program_id;
+      }
+      if (servicePoints.length === 0) {
+        console.log(`⚠ Warning: Service point with domain=${domain} not found!`);
+        return state;
+      }
+      console.log(`ℹ Service point with domain=${domain} found!`);
+      const nextState = composeNextState(state, servicePoints);
+      return callback(nextState);
+    } catch (err) {
+      console.log(`✗ Error: ${err}`);
+      return state;
+    }
+  }
+}
+
+/**
+ * get area by id in OpenSPP
+ * @public
+ * @example
+ * getArea("LOC_7M92NLDH")
+ * @function
+ * @param {string} spp_id - spp_id of area
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function getArea(spp_id, callback=(s) => s) {
+  return async state => {
+    try {
+      const area = await sppConnector.searchRead(
+        'spp.area',
+        [['spp_id', '=', spp_id]],
+        [
+          'parent_id',
+          'name',
+          'code',
+          'altnames',
+          'area_level',
+          'kind'
+        ],
+      );
+      if (area.length === 0) {
+        console.log(`⚠ Warning: Area ${spp_id} not found!`);
+        return state;
+      }
+      console.log(`ℹ Area ${spp_id} found!`);
+      const nextState = composeNextState(state, area);
+      return callback(nextState);
+    } catch (err) {
+      console.log(`✗ Error: ${err}`);
+      return state;
+    }
+  }
+}
+
+/**
+ * searching for service point in OpenSPP
+ * @public
+ * @example <caption>search without offset</caption>
+ * searchArea([["code", "=", "10732"]])
+ * @example <caption>search with offset</caption>
+ * searchArea([["kind", "=", 1]], 1)
+ * @function
+ * @param {Array} domain - searching domain
+ * @param {number} [offset=0] - offset searching
+ * @param {function} callback - An optional callback function
+ * @returns {Operation}
+ */
+export function searchArea(domain, offset=0, callback=(s) => s) {
```

Comment:
Oh, this should take an `options` object too!

Reply:
@josephjclark done 😄